osu.Game/Online/OnlineViewContainer.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Online.API;
+using osu.Game.Online.Placeholders;
+
+namespace osu.Game.Online
+{
+    /// <summary>
+    /// A <see cref="Container"/> for dislaying online content who require a local user to be logged in.
+    /// Shows its children only when the local user is logged in and supports displaying a placeholder if not.
+    /// </summary>
+    public abstract class OnlineViewContainer : VisibilityContainer, IOnlineComponent
+    {
+        private readonly Placeholder placeholder;
+        protected readonly LoadingAnimation LoadingAnimation;
+
+        private const int transform_time = 300;
+
+        private readonly Container content;
+        protected override Container<Drawable> Content => content;
+
+        [Resolved]
+        protected IAPIProvider API { get; private set; }
+
+        protected OnlineViewContainer(string placeholderMessage)
+        {
+            InternalChildren = new Drawable[]
+            {
+                content = new Container
+                {
+                    RelativeSizeAxes = Axes.Both,
+                },
+                placeholder = new LoginPlaceholder(placeholderMessage),
+                LoadingAnimation = new LoadingAnimation
+                {
+                    Alpha = 0,
+                }
+            };
+        }
+
+        public virtual void APIStateChanged(IAPIProvider api, APIState state)
+        {
+            switch (state)
+            {
+                case APIState.Offline:
+                    PopOut();
```

Comment:
That's a very dangerous way to do it (visibility state is not updated), you should be calling `Hide()` and `Show()` instead.